component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/smsotp/SMSOTPAuthenticator.java

```diff
@@ -126,7 +135,8 @@ public boolean canHandle(HttpServletRequest request) {
         return ((StringUtils.isNotEmpty(request.getParameter(SMSOTPConstants.RESEND))
                 && StringUtils.isEmpty(request.getParameter(SMSOTPConstants.CODE)))
                 || StringUtils.isNotEmpty(request.getParameter(SMSOTPConstants.CODE))
-                || StringUtils.isNotEmpty(request.getParameter(SMSOTPConstants.MOBILE_NUMBER)));
+                || StringUtils.isNotEmpty(request.getParameter(SMSOTPConstants.MOBILE_NUMBER))
+                || StringUtils.isNotEmpty(request.getParameter(SMSOTPConstants.USER_NAME)));
```

Comment:
```suggestion
                || (StringUtils.isNotEmpty(request.getParameter(EmailOTPAuthenticatorConstants.USER_NAME))
                && StringUtils.isEmpty(request.getParameter(EmailOTPAuthenticatorConstants.PASSWORD))));
```

To fix https://github.com/wso2/product-is/issues/17072

Reply:
Fixed with https://github.com/wso2-extensions/identity-outbound-auth-sms-otp/pull/178/commits/33a6880309caad5b43f77612178eada5e01093df